### DISCUSSION TITLE
Base template for multi-entity

### DISCUSSION THREAD

**Discussion by Gprotin** [gregsullivan/_tw#141] [General]
Hello 

Thank you for your hard work.

I want to use _tw as a base for a theme of multiple clients with some basic style change between them.

What would be the best option ? 

1. Developing my base theme, and bundle it with a custom theme.json for each client
2. Adding a server-side script that rebundle it based on theme.json modification

The second option would be the best for me, as I would be able to update the theme in the future without having to redeploy a large number of themes.

Thank you for your answer.
Kind regard

**Comment by gregsullivan** [comment]
Hello—Thanks for posting this!

A long time ago, when I used _s, I had a handful of changes I made every time I started a new project. I kept a list, and after generating a new theme, I would manually make those changes because it only took ten or so minutes.

Eventually, the number of changes I made started to grow, so I forked the [_s generator](https://github.com/Automattic/underscores.me). This was also a long time ago, so at the time the generator was still being updated. (It hasn't been in seven years.) I ran it locally, so I would go to something like underscores.test on my own machine, and it would produce a version of the theme with my changes. I would keep it updated as _s evolved over time.

You could do something similar. The one downside for you is that the code for the _tw generator is not public because it's based on very old code, and I haven't overhauled it to the point that I'm comfortable sharing it publicly.

That said, you could use the _s generator code as a base if you wanted to go that route, or you could create a similar generator or script from scratch. A shell script that made some minor modifications to a theme generated by underscoretw.com could also potentially solve your problem, and that's the sort of thing an AI code generator would do a pretty solid job of putting together. And since it never would run in production, it's pretty low stakes.

Hopefully that helps—good luck with your projects!

**Comment by Gprotin** [comment]
Hello,
So I have tested different approach and I think I've found the best the one.
## What I wanted
- A theme that will be used for a multiple clients
- That theme will be improved by time and deployed to all clients
- Each client have different colors, fonts... that they can change autonomously with a custom theme option page
- No intervention for each client when the parent theme is updated
## What I have tested
- A _tw parent theme with a light tailwind installation that override parent value, when the client change a value it triggers a tailwind compile and update theme.json
- Only update theme.json manually with custom value. This was not dynamic. and the child theme.json erase parent theme.json value
## What works best
- A _tw parent theme AS-IS, with wordpress variable as tailwind theme values
- A very light child theme with no css
- A php function that filter and edit parent theme.json value [WordPress Docs](https://developer.wordpress.org/news/2023/07/how-to-modify-theme-json-data-using-server-side-filters/)
### Advantages
- The client can autonomously change some tailwind variables without any compilation
- You can customize some templates for one client only directly in the child theme
- Parent theme update can be made without manual intervention and keeping customs modifications

**Reply by gregsullivan** [reply]
I really like the approach you've landed on! Now that _tw uses CSS variables to pull in the values from `theme.json`, it opens up avenues like the one you described above. (_tw used to compile the values from `theme.json` into `style.css`, but starting with Tailwind 4, we're using CSS variables instead.)
